nextest.toml

```
@@ -4,6 +4,7 @@ dir = "target/nextest"
 [profile.ci]
 failure-output = "immediate-final"
 slow-timeout = { period = "60s", terminate-after = 1 }
+retries = { backoff = "exponential", count = 3, delay = "5s" }
```

Why use an exponential back off here?
For flaky tests, we probably just want to retry them.
Also, not sure if any delay is useful at all.
Is there an argument for not retrying right away?

```suggestion
retries = { count = 3 }
```

---

yeah, I didn’t notice that we have a delay requirement. I kind of agree that we can just try retries and improve it later! btw, we’ve already retried at the Buildkite level. do we still need this framework-level retry?

---

Well the idea here is that some tests fail because of global system resource exhaustion (ports, mostly), some others fail because of the thread scheduling. Having exponential backoff gives a bit of time for those resources to be released. Though I agree that specific values could need some tuning.

As for framework-level retry, it is a quesiton of probabilities - if one test out of 100 fails, currently we restart an entire batch of 100 tests, at which point another flaky test might fail. This can take far longer than the time needed to just re-run the one flaky test, and, most importantly, it does not give the opportunity for the tests that have already passed to flake.

---

I am not against retrying.
I was just not sure that we need to wait, and even less certain we need to wait with an exponential back-off.

I suggest we retry right away.
And maybe even increase the retry `count` to a value higher than 3.